File: OrdenacaoRapida.py
import time
import tracemalloc

def ler_numeros(arquivo):
    """
    Lê linhas de 'arquivo', tenta converter em float e retorna uma lista de floats.
    Linhas vazias ou não numéricas são ignoradas.
    """
    numeros = []
    with open(arquivo, 'r') as f:
        for linha in f:
            linha = linha.strip()
            if not linha:
                continue
            try:
                numeros.append(float(linha))
            except ValueError:
                # ignora linhas que não são números válidos
                pass
    return numeros

def escrever_numeros(arr, arquivo):
    """
    Grava cada valor de arr em uma linha no arquivo de saída.
    """
    with open(arquivo, 'w') as f:
        for num in arr:
            f.write(f"{num}\n")

def main():
    entrada = "arq.txt"
    saida   = "arq-saida-python-OR.txt"

    # Leitura
    numeros = ler_numeros(entrada)

    # Inicia medição de memória
    tracemalloc.start()

    # Inicia medição de tempo
    inicio = time.perf_counter()

    # Ordenação nativa (Timsort em CPython)
    numeros.sort()

    # Finaliza medição de tempo
    fim = time.perf_counter()
    tempo_ms = (fim - inicio) * 1000  # em milissegundos

    # Pega pico de memória em bytes e converte para KB
    _, pico_bytes = tracemalloc.get_traced_memory()
    tracemalloc.stop()
    memoria_kb = pico_bytes / 1024

    # Grava saída
    escrever_numeros(numeros, saida)

    # Saída de status
    print(f"Tempo de execução: {tempo_ms:.2f} ms")
    print(f"Memória utilizada (pico): {memoria_kb:.2f} KB")
    print("Informações do PC: Intel i5-13, RAM 16GB DDR5, RTX 4050, SSD M2")
    print(f"Arquivo de saída '{saida}' gerado com sucesso.")

if __name__ == "__main__":
    main()

File: OrdenacaoRapida.java
import java.io.*;
import java.util.ArrayList;
import java.util.Arrays;
import java.lang.management.ManagementFactory;
import java.lang.management.MemoryPoolMXBean;
import java.lang.management.MemoryUsage;

public class OrdenacaoRapida {

    
    private static ArrayList<Double> lerNumeros(String arquivo) {
        ArrayList<Double> numeros = new ArrayList<>();
        try (BufferedReader br = new BufferedReader(new FileReader(arquivo))) {
            String linha;
            while ((linha = br.readLine()) != null) {
                linha = linha.trim();
                if (!linha.isEmpty()) {
                    try {
                        numeros.add(Double.parseDouble(linha));
                    } catch (NumberFormatException e) {
                    }
                }
            }
        } catch (IOException e) {
            System.err.println("Erro ao ler arquivo: " + e.getMessage());
        }
        return numeros;
    }

    private static void escreverNumeros(double[] arr, String arquivo) {
        try (BufferedWriter bw = new BufferedWriter(new FileWriter(arquivo))) {
            for (double num : arr) {
                bw.write(Double.toString(num));
                bw.newLine();
            }
        } catch (IOException e) {
            System.err.println("Erro ao escrever arquivo: " + e.getMessage());
        }
    }

    public static void main(String[] args) {
        ArrayList<Double> numeros = lerNumeros("arq.txt");
        
        double[] arr = new double[numeros.size()];
        for (int i = 0; i < arr.length; i++) {
            arr[i] = numeros.get(i);
        }

        long inicio = System.currentTimeMillis();
        
        Arrays.sort(arr);

        long tempo = System.currentTimeMillis() - inicio;

        long memoriaKB = 0;
        for (MemoryPoolMXBean pool : ManagementFactory.getMemoryPoolMXBeans()) {
            if (pool.getName().contains("Heap")) {
                MemoryUsage peakUsage = pool.getPeakUsage();
                memoriaKB = peakUsage.getUsed() / 1024;
                break;
            }
        }

        escreverNumeros(arr, "arq-saida-java-OR.txt");

        System.out.println("Tempo de execução: " + tempo + " ms");
        System.out.println("Memória utilizada: " + memoriaKB + " KB");
        System.out.println("Informacoes do PC: Intel i5-13, RAM 16GB DDR5, RTX 4050, SSD M2");
        System.out.println("Arquivo de saída 'arq-saida-java-OR.txt' gerado com sucesso.");
    }
}
